packages/typst.node/src/compiler/boxed.rs

```diff
@@ -108,35 +112,28 @@ impl BoxedCompiler {
             }
         };
 
-        let Some(new_state) = new_state else {
-            return Ok(None);
-        };
-
-        world
-            .mutate_entry(new_state)
-            .map(Some)
-            .map_err(map_node_error)
+        Ok(self.universe.snapshot_with(Some(TaskInputs {
+            entry: new_state,
+            ..Default::default()
+        })))
     }
 
     pub fn compile_raw(
         &mut self,
         compile_by: CompileDocArgs,
     ) -> napi::Result<NodeTypstCompileResult, NodeError> {
-        let e = self.setup_compiler_by(compile_by)?;
+        let world = self.setup_compiler_by(compile_by)?;
 
         let res = if self.0.universe().entry_state().is_inactive() {
             Err(map_node_error(error_once!("entry file is not set")))
         } else {
-            Ok(self.0.compile(&mut CompileEnv::default()).into())
+            Ok(self
+                .0
+                .compiler
+                .compile(&world, &mut CompileEnv::default())
```

Comment:
call into the internal method directly, because we don't want to touch the driver wrapper with a default world.

Reply:
could we name the `self.0.compiler` so to make it slightly more beautiful?
```rs
let c = &self.0.compiler;
c.compile(&world, &mut CompileEnv::default())
```